src/containers/Practice.js

```diff
@@ -0,0 +1,79 @@
+import React, {Component} from 'react'
+import {View, Text} from 'react-native'
+import {connect} from 'react-redux'
+import {PRACTICE_MODE_SELECTION, PRACTICING} from "../reducers/practice_reducer";
+import {startPractice, startPracticeMode, typeInput, eraseInput, submitTrial} from "../actions/practice_actions";
+import {PracticeModeSelection} from "../components/practice/PracticeModeSelection";
+import {Game} from "../components/game/Game";
+
+const mapStateToProps = (state) => {
+    return {
+        state: state.practice.state,
+        currentTrial: state.practice.currentTrial,
+        lastAnswerData: state.practice.lastAnswerData,
+    }
+};
+
+const mapDispatchToProps = dispatch => {
+    return {
+        actions: {
+            startPractice: () => {
+                dispatch(startPractice())
+            },
+            startPracticeMode: (category, difficulty) => {
+                dispatch(startPracticeMode(category, difficulty))
+            },
+            typeInput: (input) => {
+                dispatch(typeInput(input))
+            },
+            eraseInput: () => {
+                dispatch(eraseInput())
+            },
+            submitTrial: () => {
+                dispatch(submitTrial())
+            },
+        }
+    }
+};
+
+class Practice extends Component {
+    static navigationOptions = {
+        title: 'Practice',
+        header: null
+    };
+
+    constructor(props) {
+        super(props);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.userEnteredResult = this.userEnteredResult.bind(this);
+    }
+
+    componentWillMount() {
+        this.props.actions.startPractice();
+    }
+
+    handleSubmit() {
+        if (this.userEnteredResult()) {
+            this.props.actions.submitTrial();
+        }
+    };
+
+    userEnteredResult() {
+        return this.props.currentTrial.currentUserInput != null;
+    }
+
+    render() {
+        if (this.props.state === PRACTICE_MODE_SELECTION) {
+            return <PracticeModeSelection onSelectPracticeMode={this.props.actions.startPracticeMode}/>
+        } else if (this.props.state === PRACTICING) {
+            return <Game state={this.props.state}
```

Comment:
@MaxiSuppes Acá podes pasar directamente PRACTICING en vez de `this.props.state` que queda un toque más declarativo :)

Reply:
@lucianolev Ciertoo, cambiado